### ISSUE TITLE
NoClassDefFoundError: RequestEnvelope$AuthInfo

### ISSUE THREAD

**Issue by RootSoft** [Grover-c13/PokeGOAPI-Java#432]
I'm trying to integrate the PokeGoAPI-Java in Android Studio but I receive a NoClassDefFoundError on GoogleUserCredentialProvider.
I've build the jar file from source and followed the readme. (git submodule upgrade, ..)

Error happens here:
`provider = new GoogleUserCredentialProvider(httpClient);`

> java.lang.NoClassDefFoundError: POGOProtos.Networking.Envelopes.RequestEnvelopeOuterClass$RequestEnvelope$AuthInfo
>                                                                           at com.pokegoapi.auth.GoogleUserCredentialProvider.<init>(GoogleUserCredentialProvider.java:121)

Any idea what might be wrong?
Is this API supported by Android so far?

**Comment by Grover-c13** [comment]
Yes several people have made Android projects, did you try a gradle build to generate the protos after the 
`git submodule update --init`

**Comment by FabianTerhorst** [comment]
or wait for the jcenter release

**Comment by RootSoft** [comment]
I did try a gradle build to generate the protos but i think i solved this issue.
Building the libraries generates 4 jar files (javadoc, sources, all and library)
I only used the library but using "PokeGOAPI-library-all-0.3.jar" solved my issues.
I still had a duplicate entry on okhttp since both the library and the Android project uses the OkHttp library which I fixed with removing

```
`compile 'com.squareup.okhttp3:okhttp:3.4.1'`
```

from my app-level build.gradle file.

If other people people are still having issues, try cleaning your project after adding the library with 

> ./gradlew clean

and rebuilding the project.
Removing the build folder after a clean can also help.
Also don't forget to add 

> multiDexEnabled true

 in your app/build.gradle to support +64k methods
